tox-offline-draft.md

```diff
@@ -5,77 +5,146 @@
-hashing function will be dropped by network clients upon receival. Thus is
-makes spamming a user unreliable and costly.
-
+DHT. The SENDER and RECEIVER must complete a certain amount of computational
+hashing before they can send a message. Messages that do not properly
+complete the hashing function will be dropped by network clients upon
+receival. Thus is makes spamming a user or the network unreliable and costly.
+
+The difficulty of the PoW is increased relative to the storage space required
+to storage messages, to ensure an attacker cannot simply fill a
+STORAGE_NODE's available message space with a single message. However, very
+small messages could game this multiplier, and thus even the minimum PoW
+difficulty must be reasonably large.
+
+Messages should also expire eventually, and thus a higher PoW difficulty
+should be required for a longer message TTL (time to live).
+        
 ## Privacy
 
 Tox does not inherently provide anonymity, however this will not expose any
 more information about the user than using the DHT in Tox already does.
-(TODO: A correlation between who is talking to whom may be possible?).
+
+A middle node can correlate which IPs are talking to each other by seeing who
+requests messages sent by whom, however one could enumerate this simply by
+watching traffic from either the SENDER or RECEIVER so this does not provide
+much new information.
 
 ## Security
 
-Messages are encrypted with the same encryption they normally are, and
-users will not be able to read the content of other user's messages unless
-intended for them.
+Messages are encrypted by the SENDER with two layers, one layer for the
+STORAGE_NODE which encrypts the metadata necessary to see whom the message is
+intended for, and another layer for the RECEIVER that includes the message
+itself. Users will not be able to read the content of other user's messages
+unless intended for them.
 
 # Technical
 
-Technical details below.
-
 ## Packet details
 
 Need a packet type for sending message, sending message TTL updates,
-retreiving messages from inbox.
+retreiving messages from STORAGE_NODEs, sending PoW requests and responses.
+
+## Packet encryption
+
+There are 2 layers of encryption for a message, first the SENDER encrypts the
+message using the agreed/shared key. (TODO: Is this the Tox per-friend key we
+already use for messaging, or is this OM specific?) 
+
+The SENDER then encrypts that message for us using the SENDER -> RECEIVER
+shared key. That way the STORAGE_NODE can receive the message, encrypt it,
+and write it to disk forgetting it exists until requested. Then once
+requested, it'll send the already encrypted message to the reciever, and
+delete the data on disk.  Invalidate the current POW, and the whole cycle
+will start over again.
+
+```
+SENDER -> DHT_Node [The Message]
+    encrypts to friend       --> [sender:DHT_node shared key [SENDER : RECEIVER  key_pair || The Message]]
```

Comment:
this line contains both layers of encryption, where it should only show the first layer